src/algorithms/tonal/audio2pitch.h

```diff
@@ -0,0 +1,77 @@
+#ifndef ESSENTIA_AUDIO2PITCH_H
+#define ESSENTIA_AUDIO2PITCH_H
+
+#include "algorithmfactory.h"
+
+namespace essentia {
+namespace standard {
+
+class Audio2Pitch : public Algorithm {
+
+  protected: 
+    Input<std::vector<Real>> _frame;
+    Output<Real> _pitch;
+    Output<Real> _pitchConfidence;
+    Output<Real> _loudness;
+    Output<int> _voiced;
+
+    Algorithm* _pitchAlgorithm;
+    Algorithm* _loudnessAlgorithm;
+    // auxiliary algorithms for FFT-based pitch
+    Algorithm* _windowing;
+    Algorithm* _spectrum;
+
+    Real _sampleRate;
+    int _frameSize;
+    Real _minFrequency;
+    Real _maxFrequency;
+    std::string _pitchAlgorithmName;
+    std::string _loudnessAlgorithmName;
+    Real _tolerance;
+    Real _pitchConfidenceThreshold;
+    Real _loudnessThreshold;
+    Real _loudnessThresholdGain;
+    
+    bool isAboveThresholds(Real pitchConfidence, Real loudness);
+
+  public:
+    Audio2Pitch() {
+      declareInput(_frame, "frame", "the input frame to analyse");
+      declareOutput(_pitch, "pitch", "detected pitch in Hz");
+      declareOutput(_pitchConfidence, "pitchConfidence", "confidence of detected pitch from 0.0 - 1.0");
+      declareOutput(_loudness, "loudness", "detected loudness in decibels");
+      declareOutput(_voiced, "voiced", "voiced frame categorization, 1 for voiced and 0 for unvoiced frame");
+    }
+
+    ~Audio2Pitch() {
+      if (_pitchAlgorithm) delete _pitchAlgorithm;
+      if (_loudnessAlgorithm) delete _loudnessAlgorithm;
+      if (_windowing) delete _windowing;
+      if (_spectrum) delete _spectrum;
+    }
+
+    void declareParameters() {
+      declareParameter("sampleRate", "sample rate of incoming audio frames", "[8000,inf)", 44100);
+      declareParameter("frameSize", "size of input frame in samples", "[1,inf)", 1024);
+      declareParameter("minFrequency", "minimum frequency to detect in Hz", "[10,20000]", 60.0);
+      declareParameter("maxFrequency", "maximum frequency to detect in Hz", "[10,20000]", 2300.0);
+      declareParameter("pitchAlgorithm", "pitch algorithm to use", "{pyin,pyin_fft}", "pyin_fft");
+      declareParameter("loudnessAlgorithm", "loudness algorithm to use", "{loudness,rms}", "rms");
+      declareParameter("weighting", "string to assign a weighting function", "{custom,A,B,C,D,Z}", "custom");
+      declareParameter("tolerance", "sets tolerance for peak detection on pitch algorithm", "[0,1]", 1.0);
+      declareParameter("pitchConfidenceThreshold", "level of pitch confidence above/below which note ON/OFF start to be considered", "[0,1]", 0.25);
+      declareParameter("loudnessThreshold", "loudness level above/below which note ON/OFF start to be considered, in decibels", "[-inf,0]", -51.0);
```

Comment:
This assumes that there is the same good default threshold for both Loudness and RMS algorithms. Just wanted to double check if this is fine.

Reply:
ok, that's a good point. I have been using RMS all time. So, I will check the results with loudness with a unittest now.

Reply:
No, it is not fine. Sorry, it is something we added in the initial definition but we never checked becasue RMS was working properly. Actually Loudness algorithm is not notmalized and cannot be mapped to decibles. So, I think LOUDNESS algorithm  can be removed. 

The same for PITCHYIN, in my preliminar tests in tests it is very slow to be run in real-time (at least in python). It is not an optimized solution when a large block is used like in this case. That's why we discard it. Maybe in cpp is working fine but I didn't test it properly. So, I think we can remove too.

Reply:
To move forward, for now I propose to remove Loudness due to the inconsistency in thresholding, but keep PitchYin for now. Add a GitHub issue asking to benchmark PitchYin option in this algorithm for real-time.